'Test for function 100 lazy_matrix_mul'
'Function to be tested: lazy_matrix_mul(matrix, matrix)'
 
>>> lazy_matrix_mul = __import__('101-lazy_matrix_mul').lazy_matrix_mul

'Matrix of integers'
>>> lazy_matrix_mul([[3, 4], [6, 5]], [[8], [7]])
array([[52]
       [83]])

'Matrix of float'
>>> lazy_matrix_mul([[3.6, -4.5], [-6, 5.3]], [[-0.8], [1.7]])
array([[-10.53],
       [ 13.81]])

>>> lazy_matrix_mul((2, 5, 6), [[8, 7]])
Traceback (most recent call last):
...
ValueError: shapes (3,) and (1,2) not aligned: 3 (dim 0) != 1 (dim 0)


>>> lazy_matrix_mul([[3, 4], [6, 5]], "Meco")
Traceback (most recent call last):
...
TypeError: ufunc 'multiply' did not contain a loop with signature matching types dtype('<U4') dtype('<U4') dtype('<U4')


>>> lazy_matrix_mul([3, 4, 6, 5], [[8, 7]])
Traceback (most recent call last):
...
ValueError: shapes (4,) and (1,2) not aligned: 4 (dim 0) != 1 (dim 0)


>>> lazy_matrix_mul([[3, 4.3, 6.5, 5]], ["Meco", 7])
Traceback (most recent call last):
...
ValueError: shapes (1,4) and (2,) not aligned: 4 (dim 1) != 2 (dim 0)

>>> lazy_matrix_mul([[]], ["Meco", 7])
Traceback (most recent call last):
...
ValueError: shapes (1,0) and (2,) not aligned: 0 (dim 1) != 2 (dim 0)

>>> lazy_matrix_mul([[3, 4.3, 6.5, 5]], [[]])
Traceback (most recent call last):
...
ValueError: shapes (1,4) and (1,0) not aligned: 4 (dim 1) != 1 (dim 0)

>>> lazy_matrix_mul([[3, -4.3], [-6.5, 5, -7]], [[4, 7]])
Traceback (most recent call last):
...
ValueError: setting an array element with a sequence.


>>> lazy_matrix_mul([[3, 4.3], [6.5, 5]], [[6, 7], [3, "Meco"]])
Traceback (most recent call last):
...
ValueError: data type must provide an itemsize